conductor/src/lib.rs

```diff
@@ -136,10 +137,30 @@ pub async fn delete(client: Client, namespace: &str, name: &str) -> Result<(), C
     let coredb_api: Api<CoreDB> = Api::namespaced(client, namespace);
     let params = DeleteParams::default();
     info!("\nDeleting CoreDB: {}", name);
-    let _o = coredb_api
+    let _ = coredb_api
         .delete(name, &params)
         .await
         .map_err(ConductorError::KubeError);
+
+    // Watch for deletion
+    let wp = WatchParams {
+        field_selector: Some(format!("metadata.name={}", name)),
+        ..WatchParams::default()
+    };
+    let stream = coredb_api
+        .watch(&wp, "0")
+        .await
+        .map_err(ConductorError::KubeError)?;
+
+    let mut pinned_stream = Box::pin(stream);
+
+    while let Some(status) = pinned_stream.try_next().await? {
+        if let WatchEvent::Deleted(_) = status {
+            info!("CoreDB {} deleted", name);
+            return Ok(());
+        }
+    }
+
```

Comment:
Adding a `WatchEvent` to conductor.  This will ensure that the function will not return unless there is an error or if the `CoreDB` CR is actually deleted.

Reply:
I think there could be an issue with this, because any waiting logic in conductor blocks the processing of other events. This means in the future event that a namespace takes awhile to delete, then this replica of conductor will stop processing all other events including new instances, update instances, and so on. That's why I think it is problematic to do "wait" style logic in conductor or operator, rather "requeue" style, so that waits instead handled like interrupts.